include/pybind11/cast.h

```diff
@@ -1338,14 +1304,15 @@ template <typename... Tuple> class type_caster<std::tuple<Tuple...>> {
         return true;
     }
 
-    static handle cast_impl(const type &, return_value_policy, handle,
+    template <typename T>
+    static handle cast_impl(T &&, return_value_policy, handle,
```

Comment:
With just the `const type &` an rvalue argument will end up at the full template version.  I suppose I could put in dual empty methods (one lvalue, one rvalue) if you like that better (though there are still things like non-const lvalue and const rvalue that would slip by).